tests/loadtests/vkloadtests.cmake

```diff
@@ -280,6 +280,7 @@ target_compile_definitions(
     vkloadtests
 PRIVATE
     $<TARGET_PROPERTY:ktx,INTERFACE_COMPILE_DEFINITIONS>
+    $<$<PLATFORM_ID:Windows>:NOMINMAX>
```

Comment:
Why make this change? It sets `NOMINMAX` for all vkloadtest sources not just `InstancedSampleBase.cpp`.

Reply:
Which is the intent.

Beside the fact that the `min`/`max` macros defined in `Windows.h` were a crime against humanity (akin to `None` by the X11 headers), compiling with a vanilla latest VS2022 install, MSVC v143 toolset and cmake args

```json
"cacheVariables": {
	"KTX_FEATURE_STATIC_LIBRARY": false,
	"KTX_FEATURE_TOOLS": true,
	"KTX_FEATURE_TESTS": true,
	"KTX_FEATURE_TOOLS_CTS": true,
	"KTX_FEATURE_LOADTEST_APPS": "OpenGL+Vulkan",
	"BASISU_SUPPORT_OPENCL": false
}
```

results in the following compiler errors. A short excerpt:

```
[build] D:\Develop\KTX-Software\other_include\assimp\material.inl(101,47): error C2589: '(': illegal token on right side of '::' [D:\Develop\KTX-Software\build\msbuild-msvc\tests\loadtests\vkloadtests.vcxproj]
[build] D:\Develop\KTX-Software\other_include\assimp\material.inl(101,47): error C2589: '(': illegal token on right side of '::' [D:\Develop\KTX-Software\build\msbuild-msvc\tests\loadtests\vkloadtests.vcxproj]
[build] D:\Develop\KTX-Software\other_include\assimp\material.inl(101,47): error C2589: '(': illegal token on right side of '::' [D:\Develop\KTX-Software\build\msbuild-msvc\tests\loadtests\vkloadtests.vcxproj]
[build] D:\Develop\KTX-Software\other_include\assimp\material.inl(101,47): error C2589: '(': illegal token on right side of '::' [D:\Develop\KTX-Software\build\msbuild-msvc\tests\loadtests\vkloadtests.vcxproj]
[build] D:\Develop\KTX-Software\other_include\assimp\material.inl(101,47): error C2589: '(': illegal token on right side of '::' [D:\Develop\KTX-Software\build\msbuild-msvc\tests\loadtests\vkloadtests.vcxproj]
[build] D:\Develop\KTX-Software\other_include\assimp\material.inl(101,47): error C2589: '(': illegal token on right side of '::' [D:\Develop\KTX-Software\build\msbuild-msvc\tests\loadtests\vkloadtests.vcxproj]
[build] D:\Develop\KTX-Software\other_include\assimp\material.inl(101,47): error C2589: '(': illegal token on right side of '::' [D:\Develop\KTX-Software\build\msbuild-msvc\tests\loadtests\vkloadtests.vcxproj]
[build]   (compiling source file '../../../../tests/loadtests/vkloadtests/TextureCubemap.cpp')
[build]   D:\Develop\KTX-Software\other_include\assimp\material.h(725,14):
[build]   This diagnostic occurred in the compiler generated function 'aiReturn aiMaterial::Get(const char *,unsigned int,unsigned int,Type *,unsigned int *) const'
[build]   
[build]   (compiling source file '../../../../tests/loadtests/vkloadtests/VulkanLoadTestSample.cpp')
[build]   D:\Develop\KTX-Software\other_include\assimp\material.h(725,14):
[build]   This diagnostic occurred in the compiler generated function 'aiReturn aiMaterial::Get(const char *,unsigned int,unsigned int,Type *,unsigned int *) const'
[build]   
[build]   (compiling source file '../../../../tests/loadtests/vkloadtests/TextureMipmap.cpp')
[build]   D:\Develop\KTX-Software\other_include\assimp\material.h(725,14):
[build]   This diagnostic occurred in the compiler generated function 'aiReturn aiMaterial::Get(const char *,unsigned int,unsigned int,Type *,unsigned int *) const'
[build]   
[build]   (compiling source file '../../../../tests/loadtests/vkloadtests/TextureArray.cpp')
[build]   D:\Develop\KTX-Software\other_include\assimp\material.h(725,14):
[build]   This diagnostic occurred in the compiler generated function 'aiReturn aiMaterial::Get(const char *,unsigned int,unsigned int,Type *,unsigned int *) const'
[build]   
[build]   (compiling source file '../../../../tests/loadtests/vkloadtests/Texture.cpp')
[build]   D:\Develop\KTX-Software\other_include\assimp\material.h(725,14):
[build]   This diagnostic occurred in the compiler generated function 'aiReturn aiMaterial::Get(const char *,unsigned int,unsigned int,Type *,unsigned int *) const'
```

There are 7-8-ish source files pulling in `material.h` directly or transitively, which in turn includes `material.inl` exhibiting the innocent `std::min` invocation tripping the compiler when `min` is a macro.

Because no code written after 1998 should be relying on the presence of said macro, and they need to be turned off in multiple source files (`TextureCubemap.cpp`, `VulkanLoadTestSample.cpp`, `TextureMipmap.cpp`, etc.) instead of playing whack-a-mole, I opted in for the global `NOMINMAX` definition.

Reply:
I have no objection to the change. However something else, that I wish to identify, has been changed that is inducing these failures. Our GitHub CI uses VS2022 with both MSVC v143 and Clang toolsets and the same CMake settings that you list. Until now has it has been building successfully. GitHubs runners are regularly updated so the VS 2022 they are running is very recent.

Reply:
I can't find any builds in CI that failed due to this problem. The runners currently have Microsoft.VisualStudio.Component.VC.Tools.x86.x64 version 17.9.34511.75. 

Did this failure only happen for you on your local machine? Is your local version newer than the above?

Reply:
I just tried a CI build on the main branch. I am getting the following errors when building the loadtests: `warning C5272: throwing an object of non-copyable type 'bad_vulkan_alloc' is non-standard`. The warning is a recent addition to VS2022. The exception is one defined in KTX-Software so we can easily fix it. I haven't yet found a sufficient explanation to determine what the fix this. This exception is derived from std::bad_alloc. I expect this means we need to provide a copy constructor for `bad_vulkan_alloc`.

I expect you will encounter this in builds on this branch too.

Reply:
I have, it's in [this](https://github.com/KhronosGroup/KTX-Software/pull/909/commits/6151f9942f336d282649ddd11c977a3b2b8ab6f9) commit.

Reply:
> I have, it's in [this](https://github.com/KhronosGroup/KTX-Software/pull/909/commits/6151f9942f336d282649ddd11c977a3b2b8ab6f9) commit.

Great. Thanks. I'll concentrate on the Emscripten issue.

Reply:
As I now understand, the failures are due to a very recent version of the VS2022 compiler. So I don't have to worry about some changes in this PR having caused the failures.